extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -130,6 +147,14 @@ export class AmpViewerIntegration {
     );
   }
 
+  /**
+   * @param {string} fragment
+   * @private
+   */
+  updateUrlWithTextFragment_(fragment) {
+    window.location.replace('#:~:' + fragment);
```

Comment:
Can we validate the fragment is applied to add additional confidence for refactoring?

Reply:
Same issue as above, sinon.js does not like stubbing/spying on `window.location`.